src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpAuthHelper.cs

```diff
@@ -171,10 +171,23 @@ public void PreAuthenticateRequest(WinHttpRequestState state, uint proxyAuthSche
             // 407-401-407-401- loop.
             if (proxyAuthScheme != 0)
             {
+                ICredentials proxyCredentials;
+                Uri proxyUri;
+                if (state.Proxy != null)
+                {
+                    proxyCredentials = state.Proxy.Credentials;
+                    proxyUri = state.Proxy.GetProxy(state.RequestMessage.RequestUri);
+                }
+                else
+                {
+                    proxyCredentials = state.DefaultProxyCredentials;
+                    proxyUri = state.RequestMessage.RequestUri;
```

Comment:
`state.RequestMessage.RequestUri` is probably not the right thing to pass in here (even though it was already being used incorectly). But the problem is that the proxy URI is calculated internally via autoproxy discover logic.  I think the right fix here is more involved.